tests/drivers/sensor/accel/src/main.c

```diff
@@ -11,72 +11,312 @@
  * @}
  */
 
-#include <zephyr/ztest.h>
+#include <zephyr/drivers/emul.h>
 #include <zephyr/drivers/sensor.h>
+#include <zephyr/emul/bmi160.h>
+#include <zephyr/math/util.h>
+#include <zephyr/ztest.h>
+
+#include <bmi160.h>
 
 struct sensor_accel_fixture {
-	const struct device *accel_spi;
-	const struct device *accel_i2c;
+	const struct device *accel_0;
+	const struct device *accel_1;
+	const struct emul *accel_emul_0;
+	const struct emul *accel_emul_1;
 };
 
-static enum sensor_channel channel[] = {
-	SENSOR_CHAN_ACCEL_X,
-	SENSOR_CHAN_ACCEL_Y,
-	SENSOR_CHAN_ACCEL_Z,
-	SENSOR_CHAN_GYRO_X,
-	SENSOR_CHAN_GYRO_Y,
-	SENSOR_CHAN_GYRO_Z,
-};
+void test_sensor_accel_basic(const struct device *dev)
+{
+	SENSOR_DATA(struct sensor_three_axis_data, data, 1);
+	struct sensor_scale_metadata scale_metadata[2];
+
+	zassert_ok(sensor_read_data(dev, SENSOR_TYPE_ACCELEROMETER, data), NULL);
+	zassert_equal(1, data->header.reading_count, NULL);
+	/* Test raw values */
+	zassert_equal(0x0001, data->readings[0].x, NULL);
+	zassert_equal(0x0689, data->readings[0].y, NULL);
+	zassert_equal(0x0d11, data->readings[0].z, NULL);
+
+	/* Get the scale metadata for both sensor types */
+	zassert_ok(sensor_get_scale(dev, SENSOR_TYPE_ACCELEROMETER, &scale_metadata[0]), NULL);
+	zassert_ok(sensor_get_scale(dev, SENSOR_TYPE_GYROSCOPE, &scale_metadata[1]), NULL);
+	zassert_equal(SENSOR_RANGE_UNITS_ACCEL_G, scale_metadata[0].range_units, NULL);
+	zassert_equal(SENSOR_RANGE_UNITS_ANGLE_DEGREES, scale_metadata[1].range_units, NULL);
+
+	/* Test converted values in SI units */
+	sensor_sample_to_three_axis_data(&scale_metadata[0], data, 0);
+	zassert_within(FLOAT_TO_FP(0.0f), data->readings[0].x, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.0f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].x));
+	zassert_within(FLOAT_TO_FP(0.1f), data->readings[0].y, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.1f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].y));
+	zassert_within(FLOAT_TO_FP(0.2f), data->readings[0].z, FLOAT_TO_FP(0.01f),
+		       "Expected %" PRIf " to be within %" PRIf " of %" PRIf,
+		       PRIf_ARG(FLOAT_TO_FP(0.2f)), PRIf_ARG(FLOAT_TO_FP(0.01f)),
+		       PRIf_ARG(data->readings[0].z));
+
+	zassert_ok(sensor_read_data(dev, SENSOR_TYPE_GYROSCOPE, data), NULL);
```

Comment:
Nit: This function is getting pretty long. This chunk could be its own function.